### DISCUSSION TITLE
What happens when deleting data-dir or statusfile?

### DISCUSSION THREAD

**Discussion by martin-hollingsworth-snke-obsolete** [rauc/rauc#1195] [Q&A] [answered]
Hello RAUC team, hello community,
I am upgrading my device from RAUC 1.5 -> 1.7 -> latest. In this process I noticed that my old system (Rauc v1.5) doesn't use a `statusfile `. In version 1.7 this doesn't seem to be a problem. In the latest version the recommendation is to use the  `data-directory`.

Question: my rootfs is RO, I have a shared data partition for RW application config data. However, the data partition will be erased when the user runs factory reset. **How does the loss of the `data-dir` / `statusfile` affect the RAUC operation?** (Maybe a short annotation in the docs about this would be usefeull?)

Thanks and have a great day!

**Comment by jluebbe** [comment, marked as answer]
You can still install new updates, but you will lose information about which bundles/images are installed and when they were installed.

If you want to keep that kind of information, you should not remove the `data-dir` during a factory reset.

**Comment by martin-hollingsworth-snke-obsolete** [comment]
I have a related follow up question, sorry for re-opening old entries:

The system runs from a read only SD card and rauc updates the partitions on the SD card. The rauc status file is placed on a different storage device (i.e. an SSD drive). If I as developer remove the SD card and manually flash a different version onto it, the rootfs on the SD card and the slot info printed in `rauc status --detailed` do not match.

Is there any way to let rauc validate that the status file still matches the rootfs?

**Reply by ejoerns** [reply]
No, there is no such check. Depending on the image type on could probably re-read the image and compare the signature, but for e.g. tar extraction this will not work.
So far RAUC assumes to be the only instance updating the file systems. If there is another, it has to take care of invalidating the status file.

**Reply by jluebbe** [reply]
Or place the status file on the SD card as well, so it is replaced together with the installed slots.
